gcc13 possibly uninitialized value warning

```bash
In file included from /usr/include/c++/13/functional:59,
                 from /__w/liblifthttp/liblifthttp/inc/lift/request.hpp:12,
                 from /__w/liblifthttp/liblifthttp/inc/lift/executor.hpp:3,
                 from /__w/liblifthttp/liblifthttp/inc/lift/client.hpp:3,
                 from /__w/liblifthttp/liblifthttp/src/client.cpp:1:
In constructor 'std::function<_Res(_ArgTypes ...)>::function(std::function<_Res(_ArgTypes ...)>&&) [with _Res = void; _ArgTypes = {std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response}]',
    inlined from 'constexpr std::__detail::__variant::_Uninitialized<_Type, false>::_Uninitialized(std::in_place_index_t<0>, _Args&& ...) [with _Args = {std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>}; _Type = std::function<void(std::unique_ptr<lift::request>, lift::response)>]' at /usr/include/c++/13/variant:282:4,
    inlined from 'constexpr std::__detail::__variant::_Variadic_union<_First, _Rest ...>::_Variadic_union(std::in_place_index_t<0>, _Args&& ...) [with _Args = {std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>}; _First = std::function<void(std::unique_ptr<lift::request>, lift::response)>; _Rest = {std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:411:4,
    inlined from 'constexpr std::__detail::__variant::_Variadic_union<_First, _Rest ...>::_Variadic_union(std::in_place_index_t<_Np>, _Args&& ...) [with long unsigned int _Np = 1; _Args = {std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>}; _First = std::monostate; _Rest = {std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:417:4,
    inlined from 'void std::_Construct(_Tp*, _Args&& ...) [with _Tp = __detail::__variant::_Variadic_union<monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >; _Args = {const in_place_index_t<1>&, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>}]' at /usr/include/c++/13/bits/stl_construct.h:119:7,
    inlined from 'std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)> mutable [with auto:4 = std::function<void(std::unique_ptr<lift::request>, lift::response)>; auto:5 = std::integral_constant<long unsigned int, 1>]' at /usr/include/c++/13/variant:632:23,
    inlined from 'constexpr _Res std::__invoke_impl(__invoke_other, _Fn&&, _Args&& ...) [with _Res = void; _Fn = __detail::__variant::_Move_ctor_base<false, monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)>; _Args = {function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, integral_constant<long unsigned int, 1>}]' at /usr/include/c++/13/bits/invoke.h:61:36,
    inlined from 'constexpr typename std::__invoke_result<_Functor, _ArgTypes>::type std::__invoke(_Callable&&, _Args&& ...) [with _Callable = __detail::__variant::_Move_ctor_base<false, monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)>; _Args = {function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, integral_constant<long unsigned int, 1>}]' at /usr/include/c++/13/bits/invoke.h:96:40,
    inlined from 'static constexpr decltype(auto) std::__detail::__variant::__gen_vtable_impl<std::__detail::__variant::_Multi_array<_Result_type (*)(_Visitor, _Variants ...)>, std::integer_sequence<long unsigned int, __indices ...> >::__visit_invoke(_Visitor&&, _Variants ...) [with _Result_type = std::__detail::__variant::__variant_idx_cookie; _Visitor = std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)>&&; _Variants = {std::variant<std::monostate, std::fu
    inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__variant_idx_cookie; _Visitor = __detail::__variant::_Move_ctor_base<false, monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)>; _Variants = {variant<monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >}]' at /usr/include/c++/13/variant:1816:5,
    inlined from 'constexpr decltype(auto) std::__do_visit(_Visitor&&, _Variants&& ...) [with _Result_type = __detail::__variant::__variant_idx_cookie; _Visitor = __detail::__variant::_Move_ctor_base<false, monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)>; _Variants = {variant<monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >}]' at /usr/include/c++/13/variant:1756:5,
    inlined from 'constexpr void std::__detail::__variant::__raw_idx_visit(_Visitor&&, _Variants&& ...) [with _Visitor = _Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<false, std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >&&)::<lambda(auto:4&&, auto:5)>; _Variants = {std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >}]' at /usr/include/c++/13/variant:183:44,
    inlined from 'std::__detail::__variant::_Move_ctor_base<<anonymous>, _Types>::_Move_ctor_base(std::__detail::__variant::_Move_ctor_base<<anonymous>, _Types>&&) [with bool <anonymous> = false; _Types = {std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:627:28,
    inlined from 'std::__detail::__variant::_Copy_assign_base<<anonymous>, _Types>::_Copy_assign_base(std::__detail::__variant::_Copy_assign_base<<anonymous>, _Types>&&) [with bool <anonymous> = false; _Types = {std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:692:7,
    inlined from 'std::__detail::__variant::_Move_assign_base<<anonymous>, _Types>::_Move_assign_base(std::__detail::__variant::_Move_assign_base<<anonymous>, _Types>&&) [with bool <anonymous> = false; _Types = {std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:746:7,
    inlined from 'std::__detail::__variant::_Variant_base<_Types>::_Variant_base(std::__detail::__variant::_Variant_base<_Types>&&) [with _Types = {std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:777:7,
    inlined from 'std::variant<_Types>::variant(std::variant<_Types>&&) [with _Types = {std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> >}]' at /usr/include/c++/13/variant:1429:7,
    inlined from 'void std::_Construct(_Tp*, _Args&& ...) [with _Tp = variant<monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >; _Args = {variant<monostate, function<void(unique_ptr<lift::request, default_delete<lift::request> >, lift::response)>, promise<pair<unique_ptr<lift::request, default_delete<lift::request> >, lift::response> > >}]' at /usr/include/c++/13/bits/stl_construct.h:119:7,
    inlined from 'constexpr void std::_Optional_payload_base<_Tp>::_M_construct(_Args&& ...) [with _Args = {std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >}; _Tp = std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >]' at /usr/include/c++/13/optional:278:19,
    inlined from 'constexpr void std::_Optional_payload_base<_Tp>::_M_move_assign(std::_Optional_payload_base<_Tp>&&) [with _Tp = std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >]' at /usr/include/c++/13/optional:198:26,
    inlined from 'constexpr void std::_Optional_payload_base<_Tp>::_M_move_assign(std::_Optional_payload_base<_Tp>&&) [with _Tp = std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >]' at /usr/include/c++/13/optional:189:7,
    inlined from 'constexpr std::_Optional_payload<_Tp, true, false, false>& std::_Optional_payload<_Tp, true, false, false>::operator=(std::_Optional_payload<_Tp, true, false, false>&&) [with _Tp = std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >]' at /usr/include/c++/13/optional:422:22,
    inlined from 'constexpr std::_Optional_payload<_Tp, false, _Copy, _Move>& std::_Optional_payload<_Tp, false, _Copy, _Move>::operator=(std::_Optional_payload<_Tp, false, _Copy, _Move>&&) [with _Tp = std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >; bool _Copy = false; bool _Move = false]' at /usr/include/c++/13/optional:438:26,
    inlined from 'constexpr std::_Optional_base<_Tp, <anonymous>, <anonymous> >& std::_Optional_base<_Tp, <anonymous>, <anonymous> >::operator=(std::_Optional_base<_Tp, <anonymous>, <anonymous> >&&) [with _Tp = std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >; bool <anonymous> = false; bool <anonymous> = false]' at /usr/include/c++/13/optional:552:23,
    inlined from 'constexpr std::optional<std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > > >& std::optional<std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > > >::operator=(std::optional<std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > > >&&)' at /usr/include/c++/13/optional:707:11,
    inlined from 'lift::request::async_future_type lift::request::async_future()' at /__w/liblifthttp/liblifthttp/inc/lift/request.hpp:541:63,
    inlined from 'lift::request::async_future_type lift::client::start_request(lift::request_ptr&&)' at /__w/liblifthttp/liblifthttp/src/client.cpp:174:45:
/usr/include/c++/13/bits/std_function.h:405:42: warning: '*(std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>*)((char*)&<unnamed> + offsetof(std::optional<std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > > >,std::optional<std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > > >::<unnamed>.std::_Optional_base<std::variant<std::monostate, std::function<void(std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response)>, std::promise<std::pair<std::unique_ptr<lift::request, std::default_delete<lift::request> >, lift::response> > >, false, false>::<unnamed
  405 |       : _Function_base(), _M_invoker(__x._M_invoker)
      |                                      ~~~~^~~~~~~~~~
/__w/liblifthttp/liblifthttp/inc/lift/request.hpp: In member function 'lift::request::async_future_type lift::client::start_request(lift::request_ptr&&)':
/__w/liblifthttp/liblifthttp/inc/lift/request.hpp:541:63: note: '<anonymous>' declared here
  541 |         m_on_complete_handler.m_object = {async_promise_type{}};
```

--- a/.githooks/pre-commit
+++ b/.githooks/pre-commit
@@ -12,7 +12,7 @@ should_clang_format() {
     local result=0
 
     # Ignore the test/catch_amalgamated.hpp|cpp files
-    if [[ "$1" != *"catch_amalgamated"* ]]; then
+    if [[ "$1" != *"catch_amalgamated"* && "$1" != *"pragma"* ]]; then
         for ext in $FILE_EXTS; do
             # Otherwise, if the extension is in the array of extensions to reformat, echo 1.
             [[ "$ext" == "$extension" ]] && result=1 && break

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -30,6 +30,7 @@ message("${PROJECT_NAME} LIFT_USER_LINK_LIBRARIES = ${LIFT_USER_LINK_LIBRARIES}"
 
 set(LIBLIFTHTTP_SOURCE_FILES
     inc/lift/impl/copy_util.hpp
+    inc/lift/impl/pragma.hpp
 
     inc/lift/client.hpp src/client.cpp
     inc/lift/const.hpp

--- a/README.md
+++ b/README.md
@@ -47,9 +47,8 @@ int main()
     // Debug information about any request can be added by including a callback handler for debug
     // information.  Just pass in a lambda to capture the verbose debug information.
     sync_request.debug_info_handler(
-        [](const lift::request& /*unused*/, lift::debug_info_type type, std::string_view data) {
-            std::cout << "sync_request (" << lift::to_string(type) << "): " << data;
-        });
+        [](const lift::request& /*unused*/, lift::debug_info_type type, std::string_view data)
+        { std::cout << "sync_request (" << lift::to_string(type) << "): " << data; });
 
     // Set the http method for this synchronous request.
     sync_request.method(lift::http::method::post);
@@ -96,7 +95,8 @@ int main()
     // Start the request that will be completed by callback.
     client.start_request(
         std::move(async_callback_request),
-        [](lift::request_ptr async_callback_request_returned, lift::response async_callback_response) {
+        [](lift::request_ptr async_callback_request_returned, lift::response async_callback_response)
+        {
             // This on complete callback will run on the lift::client background event loop thread.
             std::cout << "Lift status (async callback): ";
             std::cout << lift::to_string(async_callback_response.lift_status()) << "\n";

--- a/examples/benchmark.cpp
+++ b/examples/benchmark.cpp
@@ -38,11 +38,11 @@ int main(int argc, char* argv[])
 {
     constexpr char   short_options[] = "c:d:t:h";
     constexpr option long_options[]  = {
-        {"help", no_argument, nullptr, 'h'},
-        {"connections", required_argument, nullptr, 'c'},
-        {"duration", required_argument, nullptr, 'd'},
-        {"threads", required_argument, nullptr, 't'},
-        {nullptr, 0, nullptr, 0}};
+         {"help", no_argument, nullptr, 'h'},
+         {"connections", required_argument, nullptr, 'c'},
+         {"duration", required_argument, nullptr, 'd'},
+         {"threads", required_argument, nullptr, 't'},
+         {nullptr, 0, nullptr, 0}};
 
     int option_index = 0;
     int opt          = 0;
@@ -109,7 +109,8 @@ int main(int argc, char* argv[])
             clients.emplace_back(std::make_unique<lift::client>());
 
             callbacks.emplace_back(
-                [&clients, &success, &error, &callbacks, i](lift::request_ptr req_ptr, lift::response response) {
+                [&clients, &success, &error, &callbacks, i](lift::request_ptr req_ptr, lift::response response)
+                {
                     if (response.lift_status() == lift::lift_status::success)
                     {
                         success.fetch_add(1, std::memory_order_relaxed);

--- a/examples/readme.cpp
+++ b/examples/readme.cpp
@@ -14,9 +14,8 @@ int main()
     // Debug information about any request can be added by including a callback handler for debug
     // information.  Just pass in a lambda to capture the verbose debug information.
     sync_request.debug_info_handler(
-        [](const lift::request& /*unused*/, lift::debug_info_type type, std::string_view data) {
-            std::cout << "sync_request (" << lift::to_string(type) << "): " << data;
-        });
+        [](const lift::request& /*unused*/, lift::debug_info_type type, std::string_view data)
+        { std::cout << "sync_request (" << lift::to_string(type) << "): " << data; });
 
     // Set the http method for this synchronous request.
     sync_request.method(lift::http::method::post);
@@ -63,7 +62,8 @@ int main()
     // Start the request that will be completed by callback.
     client.start_request(
         std::move(async_callback_request),
-        [](lift::request_ptr async_callback_request_returned, lift::response async_callback_response) {
+        [](lift::request_ptr async_callback_request_returned, lift::response async_callback_response)
+        {
             // This on complete callback will run on the lift::client background event loop thread.
             std::cout << "Lift status (async callback): ";
             std::cout << lift::to_string(async_callback_response.lift_status()) << "\n";

--- a/inc/lift/executor.hpp
+++ b/inc/lift/executor.hpp
@@ -34,10 +34,10 @@ class executor
     friend request;
 
 public:
-    executor(const executor&) = delete;
-    executor(executor&&)      = delete;
+    executor(const executor&)                    = delete;
+    executor(executor&&)                         = delete;
     auto operator=(const executor&) -> executor& = delete;
-    auto operator=(executor&&) -> executor& = delete;
+    auto operator=(executor&&) -> executor&      = delete;
     ~executor();
 
 private:

--- a/inc/lift/http.hpp
+++ b/inc/lift/http.hpp
@@ -1,7 +1,7 @@
 #pragma once
 
-#include <string>
 #include <cstdint>
+#include <string>
 
 #include <curl/curl.h>
 

--- a/inc/lift/impl/pragma.hpp
+++ b/inc/lift/impl/pragma.hpp
@@ -1,0 +1,19 @@
+#if defined(__GNUC__) && !defined(__llvm__)
+    #define DO_PRAGMA(X)                 _Pragma(#X)
+    #define DISABLE_WARNING_PUSH         DO_PRAGMA(GCC diagnostic push)
+    #define DISABLE_WARNING_POP          DO_PRAGMA(GCC diagnostic pop)
+    #define DISABLE_WARNING(warningName) DO_PRAGMA(GCC diagnostic ignored #warningName)
+
+    #define DISABLE_WARNING_MAYBE_UNINITIALIZED DISABLE_WARNING(-Wmaybe-uninitialized)
+#elif defined(__clang__)
+    #define DO_PRAGMA(X) _Pragma(#X)
+    #define DISABLE_WARNING_PUSH
+    #define DISABLE_WARNING_POP
+    #define DISABLE_WARNING(warningNumber)
+
+    #define DISABLE_WARNING_MAYBE_UNINITIALIZED
+#else
+    // unknown compiler, ignoring suppression directives
+    #define DISABLE_WARNING_PUSH
+    #define DISABLE_WARNING_POP
+#endif

--- a/inc/lift/lift_status.hpp
+++ b/inc/lift/lift_status.hpp
@@ -1,7 +1,7 @@
 #pragma once
 
-#include <string>
 #include <cstdint>
+#include <string>
 
 namespace lift
 {

--- a/inc/lift/query_builder.hpp
+++ b/inc/lift/query_builder.hpp
@@ -1,10 +1,10 @@
 #pragma once
 
+#include <cstdint>
 #include <sstream>
 #include <string>
 #include <string_view>
 #include <vector>
-#include <cstdint>
 
 namespace lift
 {

--- a/inc/lift/request.hpp
+++ b/inc/lift/request.hpp
@@ -3,6 +3,7 @@
 #include "lift/header.hpp"
 #include "lift/http.hpp"
 #include "lift/impl/copy_util.hpp"
+#include "lift/impl/pragma.hpp"
 #include "lift/mime_field.hpp"
 #include "lift/resolve_host.hpp"
 #include "lift/response.hpp"
@@ -155,10 +156,10 @@ public:
         return std::make_unique<request>(std::move(url), std::move(timeout));
     }
 
-    request(const request&) = default;
-    request(request&&)      = default;
+    request(const request&)                             = default;
+    request(request&&)                                  = default;
     auto operator=(const request&) noexcept -> request& = default;
-    auto operator=(request&&) noexcept -> request& = default;
+    auto operator=(request&&) noexcept -> request&      = default;
 
     /**
      * Synchronously executes this request.
@@ -538,7 +539,12 @@ private:
      */
     auto async_future() -> async_future_type
     {
+        // gcc-13 is incorrectly thinking the 2nd type in the async_handlers_type is uninitialized.
+        // Its correct since it isn't used on this code path, but it isn't a warning/error.
+        DISABLE_WARNING_PUSH
+        DISABLE_WARNING_MAYBE_UNINITIALIZED
         m_on_complete_handler.m_object = {async_promise_type{}};
+        DISABLE_WARNING_POP
         return std::get<async_promise_type>(m_on_complete_handler.m_object.value()).get_future();
     }
 

--- a/inc/lift/resolve_host.hpp
+++ b/inc/lift/resolve_host.hpp
@@ -1,7 +1,7 @@
 #pragma once
 
-#include <string>
 #include <cstdint>
+#include <string>
 
 namespace lift
 {
@@ -22,9 +22,9 @@ public:
     resolve_host(std::string resolve_host, uint16_t resolve_port, std::string resolved_ip_addr);
     ~resolve_host() = default;
 
-    resolve_host(const resolve_host&)     = default;
-    resolve_host(resolve_host&&) noexcept = default;
-    auto operator=(const resolve_host&) -> resolve_host& = default;
+    resolve_host(const resolve_host&)                        = default;
+    resolve_host(resolve_host&&) noexcept                    = default;
+    auto operator=(const resolve_host&) -> resolve_host&     = default;
     auto operator=(resolve_host&&) noexcept -> resolve_host& = default;
 
     /**

--- a/inc/lift/response.hpp
+++ b/inc/lift/response.hpp
@@ -27,10 +27,10 @@ public:
     response();
     ~response() = default;
 
-    response(const response&) = default;
-    response(response&&)      = default;
+    response(const response&)                             = default;
+    response(response&&)                                  = default;
     auto operator=(const response&) noexcept -> response& = default;
-    auto operator=(response&&) noexcept -> response& = default;
+    auto operator=(response&&) noexcept -> response&      = default;
 
     /**
      * The Lift status is how the request ended up in the event loop.

--- a/inc/lift/share.hpp
+++ b/inc/lift/share.hpp
@@ -42,10 +42,10 @@ public:
     explicit share(options opts);
     ~share();
 
-    share(const share&) = delete;
-    share(share&&)      = delete;
+    share(const share&)                             = delete;
+    share(share&&)                                  = delete;
     auto operator=(const share&) noexcept -> share& = delete;
-    auto operator=(share&&) noexcept -> share& = delete;
+    auto operator=(share&&) noexcept -> share&      = delete;
 
     static auto make_shared(options opts) -> std::shared_ptr<share> { return std::make_shared<share>(std::move(opts)); }
 

--- a/src/executor.cpp
+++ b/src/executor.cpp
@@ -73,6 +73,8 @@ auto executor::prepare() -> void
 
     switch (m_request->method())
     {
+        default:
+            /* INTENTIONAL FALLTHROUGH */
         case http::method::unknown: // default to GET on unknown/bad value.
             /* INTENTIONAL FALLTHROUGH */
         case http::method::get:
@@ -103,6 +105,8 @@ auto executor::prepare() -> void
 
     switch (m_request->version())
     {
+        default:
+            /* INTENTIONAL FALLTHROUGH */
         case http::version::unknown: // default to USE_BEST on unknown/bad value.
             /* INTENTIONAL FALLTHROUGH */
         case http::version::use_best:
@@ -223,6 +227,9 @@ auto executor::prepare() -> void
             {
                 switch (auth_type)
                 {
+                    default:
+                        // CURLAUTH_BASIC is the default per docs https://curl.se/libcurl/c/CURLOPT_PROXYAUTH.html.
+                        /* INTENTIONAL FALLTHROUGH */
                     case http_auth_type::basic:
                         auth_types |= CURLAUTH_BASIC;
                         break;
@@ -536,10 +543,9 @@ auto curl_write_data(void* buffer, size_t size, size_t nitems, void* user_ptr) -
     auto&  response     = executor_ptr->m_response;
     size_t data_length  = size * nitems;
 
-    std::copy(
-        static_cast<const char*>(buffer),
-        static_cast<const char*>(buffer) + data_length,
-        std::back_inserter(response.m_data));
+    std::string_view from{static_cast<const char*>(buffer), data_length};
+
+    std::copy(from.begin(), from.end(), std::back_inserter(response.m_data));
 
     return data_length;
 }

